### PR TITLE
Constants: Change managing of user data

### PR DESCRIPTION
**Pull request by Uran198**
Replace manual management of configuration and user data storage
with appdirs.

Fixes https://github.com/coala-analyzer/coala/issues/1805

### CODE REVIEW

**Comment by sils1297** [comment]
hey, code is good, he commit message needs to be changed a bit, *how* do you change it? Every commit is obviously a change.

**Comment by Uran198** [comment]
@sils1297 You comment sounds a little bit like a riddle :smile:

**Comment by sils1297** [comment]
from gitter:

@Uran198 yeah I'm sometimes a bit riddly when it comes to commit messages
@Uran198 can you try coming up with something better? I'm deliberately not giving you the exact info because I want to get you to think about it so you learn it. So intentions are good FWIW : )

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/1901/commits/1e9f82468b0ecb3f12b3496b11dbe1f97cc6f082

**Comment by sils1297** [comment]
@rultor merge

File: requirements.txt [hunk 1]
```diff
@@ -3,3 +3,4 @@ setuptools>=19.2
 # Do *not* use 0.3 which is not backwards compatible to common clang versions
 libclang-py3==0.2
 validators>=0.10
+appdirs==1.4.0
```
Comment (sils1297) [inline requirements.txt]: oh btw. can you use `~=`?

**Comment by sils1297** [comment]
unack 1e9f824

**Comment by sils1297** [comment]
Oh we have a bug in our rultor config, we currently install deps in the docker image, I've added appdirs there now. Can you file an issue that we should install deps in rultor and not in the image so they are synched with our requirements.txt?

https://hub.docker.com/r/coala/rultor-python/builds/ , when that finishes we should be able to merge.

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala/pull/1901/commits/e5db5655a6ae65ae74c7f56f878bd903a9433893 ack

**Comment by AbdealiJK** [comment]
It seems travis failed. So im makring it as WIP
(Ive cancelled the other CIs - so you can ignore their failure)

**Comment by AbdealiJK** [comment]
Also, rebase needed

**Comment by AbdealiJK** [comment]
Travis failed again ?
Seems to be something about `~=`

**Comment by sils1297** [comment]
we never used that before but it *should* work right?

**Comment by AbdealiJK** [comment]
yes. i tested it locally.

Maybe travis has an old pip ?

**Comment by sils1297** [comment]
is this functionality rather new?

Sincerely,

Lasse Schuirmann

contact@viperdev.io
http://viperdev.io/

2016-03-17 15:02 GMT+01:00 AbdealiJK <notifications@github.com>:

**Comment by Uran198** [comment]
Support for PEP440 was implemented in setuptools v8.0 and pip v6.0. From https://packaging.python.org/en/latest/glossary/#term-version-specifier

**Comment by Uran198** [comment]
And here is similar problem, that was caused by outdated pip: http://stackoverflow.com/questions/28502283/valueerror-expected-version-spec-when-installing-local-wheel-via-pip

**Comment by AbdealiJK** [comment]
You could try `pip install -U pip` in travis and see if it gets fixed ?

Cuz appveyor and circle dont complain.

**Comment by sils1297** [comment]
ack a838631 331406c

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**